corehq/apps/case_search/utils.py

```diff
@@ -102,4 +105,17 @@ def _add_case_property_queries(self):
             for removal_regex in remove_char_regexs:
                 to_remove = re.escape(removal_regex.regex)
                 value = re.sub(to_remove, '', value)
-            self.search_es = self.search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+            if self.wildcard_enabled:
+                values = value.split()
+                if len(values) == 1 and key in fuzzies:
+                    # fuzzy search can only be supported for single words
+                    self.search_es = self.search_es.case_property_query(
+                        key, value, fuzzy=True, clause=queries.SHOULD)
+                    clause = queries.SHOULD
+                else:
+                    clause = queries.MUST
+                for token in values:
+                    token = ".*" + token.lower() + ".*"
```

Comment:
According to [this](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-wildcard-query.html) adding this to the beginning and end of your search term is going to be particularly bad for performance. I don't know if the mechanism for the regexp queries are the same as for a wildcard, but I would test this code out on a sample domain with lots of cases  (e.g. the `polar-bears-are-cool` domain on staging) to make sure that this isn't going completely crush Elasticsearch.

Reply:
Please see above [comment](https://github.com/dimagi/commcare-hq/pull/29010/files#r562478475)